hono/src/services/items.ts

```diff
@@ -23,8 +23,28 @@ const selectItemSchema = createSelectSchema(items, {
   id: z.string().refine((val) => isCuid(val)),
 })
 
-app.get('/', injectDB, async (c) => {
-  return c.json(await c.get('db').query.items.findMany())
+const paginationValidation = z.object({
+  page: z
+    .string()
+    .refine((val) => !isNaN(+val) && +val >= 0, {
+      message: 'Page must be a non-negative number',
+    })
+    .optional(),
+  size: z
+    .string()
+    .refine((val) => !isNaN(+val) && +val > 0, {
+      message: 'Size must be a positive number',
+    })
+    .optional(),
+})
+
+app.get('/', zValidator('query', paginationValidation), injectDB, async (c) => {
+  const { page, size } = c.req.query()
+
+  const pageNumber: number = page ? +page : 0
+  const pageSize: number = size ? +size : 25
+
+  return c.json(await c.get('db').select().from(items).limit(pageSize).offset(pageNumber))
```

Comment:
This wasn't in the initial acceptance criteria for this issue, but it is commonly best practice to give the consumer a "total count" metric so they can understand how to use pagination. The response could look like:
```json
{
    "items": [
        { ... }
    ],
    "total": 106
}
```
When calculating the total, make sure to look into implementation details. There is are fast ways and multiple slow ways to find the total number of records in a db. Make sure you are filtering by authorUsername hardcoded to "rak3rman" (shouldn't matter for now but will matter soon).

- [ ] Update issue #98 to contain this additional constraint
- [ ] Implement the count metric/calculation

Reply:
This metric has been implemented. The current solution was opted instead of the solution below because it only calls the table `items` once.

```JS
const res = await c
    .get('db')
    .select()
    .from(items)
    .where(eq(items.authorUsername, 'rak3rman'))
    .limit(pageSize)
    .offset(pageNumber)

const total = (
    await c.get('db').select({ total: count() }).from(items).where(eq(items.authorUsername, 'rak3rman'))
  )[0].total
  ```

Reply:
Comment outdated, base version has been implemented.